core/types/transaction.go

```diff
@@ -510,11 +540,19 @@ func NewTxWithMinerFee(tx *Transaction, baseFee *big.Int) (*TxWithMinerFee, erro
 func NewBundleWithMinerFee(bundle *SimulatedBundle, baseFee *big.Int) (*TxWithMinerFee, error) {
 	minerFee := bundle.MevGasPrice
 	return &TxWithMinerFee{
-		order:    _BundleOrder{bundle},
+		order:    SimulatedBundleOrder{Bundle: bundle},
 		minerFee: minerFee,
 	}, nil
 }
 
+func NewOrderWithTxFee(order Order, baseFee *big.Int) (*TxWithMinerFee, error) {
+	if order.TxType() == Tx {
+		return NewTxWithMinerFee(order.AsTx(), baseFee)
+	} else {
+		return NewBundleWithMinerFee(order.AsSimulatedBundle(), baseFee)
```

Comment:
order being a regular bundle would just break everything, the interfaces are incompatible I think